.pre-commit-config.yaml

```
@@ -0,0 +1,10 @@
+# See https://pre-commit.com for more information
+# See https://pre-commit.com/hooks.html for more hooks
+repos:
+  - repo: local
+    hooks:
+      - id: black
+        name: black
+        entry: black
+        language: system
+        types: [python]
```

I'm curious why you use local here rather than the [reusable repo](https://stackoverflow.com/a/67796237) e.g.
```suggestion
  - repo: https://github.com/psf/black
    rev: 24.4.0
    hooks:
    - id: black
      language_version: python3
```

---

I'm just going to leave it as is so that we don't have to pin it to a specific 'rev' version. Well see how it goes